File: app/src/main/java/com/kelsos/mbrc/adapters/LyricsAdapter.java
package com.kelsos.mbrc.adapters;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.kelsos.mbrc.R;
import java.util.ArrayList;

public class LyricsAdapter extends RecyclerView.Adapter<LyricsAdapter.ViewHolder> {
  private ArrayList<String> mData;
  private Typeface robotoLight;

  public LyricsAdapter(Context context, ArrayList<String> objects) {
    this.mData = objects;
    robotoLight = Typeface.createFromAsset(context.getAssets(), "fonts/roboto_light.ttf");
  }

  /**
   * Called when RecyclerView needs a new {@link ViewHolder} of the given type to represent
   * an item.
   * <p/>
   * This new ViewHolder should be constructed with a new View that can represent the items
   * of the given type. You can either create a new View manually or inflate it from an XML
   * layout file.
   * <p/>
   * The new ViewHolder will be used to display items of the adapter using
   * {@link #onBindViewHolder(ViewHolder, int)}. Since it will be re-used to display different
   * items in the data set, it is a good idea to cache references to sub views of the View to
   * avoid unnecessary {@link View#findViewById(int)} calls.
   *
   * @param parent The ViewGroup into which the new View will be added after it is bound to
   * an adapter position.
   * @param viewType The view type of the new View.
   * @return A new ViewHolder that holds a View of the given view type.
   * @see #getItemViewType(int)
   * @see #onBindViewHolder(ViewHolder, int)
   */
  @Override public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    View view = LayoutInflater.from(parent.getContext())
        .inflate(R.layout.ui_list_lyrics_item, parent, false);
    return new ViewHolder(view, robotoLight);
  }

  /**
   * Called by RecyclerView to display the data at the specified position. This method
   * should update the contents of the {@link ViewHolder#itemView} to reflect the item at
   * the given position.
   * <p/>
   * Note that unlike {@link android.widget.ListView}, RecyclerView will not call this
   * method again if the position of the item changes in the data set unless the item itself
   * is invalidated or the new position cannot be determined. For this reason, you should only
   * use the <code>position</code> parameter while acquiring the related data item inside this
   * method and should not keep a copy of it. If you need the position of an item later on
   * (e.g. in a click listener), use {@link ViewHolder#getPosition()} which will have the
   * updated position.
   *
   * @param holder The ViewHolder which should be updated to represent the contents of the
   * item at the given position in the data set.
   * @param position The position of the item within the adapter's data set.
   */
  @Override public void onBindViewHolder(ViewHolder holder, int position) {
    String str = mData.get(position);
    holder.title.setText(str);
  }

  /**
   * Returns the total number of items in the data set hold by the adapter.
   *
   * @return The total number of items in this adapter.
   */
  @Override public int getItemCount() {
    return mData == null ? 0 : mData.size();
  }

  public static class ViewHolder extends RecyclerView.ViewHolder {
    TextView title;

    public ViewHolder(View itemView, Typeface typeface) {
      super(itemView);
      title = (TextView) itemView.findViewById(android.R.id.text1);
      title.setTypeface(typeface);
    }
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/adapters/LyricsAdapter.kt
package com.kelsos.mbrc.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.kelsos.mbrc.R

class LyricsAdapter(private var data: List<String> = emptyList()) : RecyclerView.Adapter<LyricsAdapter.ViewHolder>() {

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
    val view = LayoutInflater.from(parent.context).inflate(R.layout.ui_list_lyrics_item, parent, false)
    return ViewHolder(view)
  }

  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
    val str = data[position]
    holder.title.text = str
  }

  fun updateLyrics(lyrics: List<String>) {
    this.data = lyrics
    notifyDataSetChanged()
  }

  override fun getItemCount(): Int {
    return data.size
  }

  class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    @BindView(android.R.id.text1) lateinit var title: TextView

    init {
      ButterKnife.bind(this, itemView)
    }
  }

  fun clear() {
    if (data.isNotEmpty()) {
      data = emptyList()
    }
  }
}
